src/model.ts

```diff
@@ -1,13 +1,30 @@
 import { Request } from 'express';
-import { Readable } from 'stream';
+import { PagingStream } from './paging-stream';
+import { ISearchApiOptions } from './types';
+import { getSearchQueryParams } from './get-search-query-params';
 
 export class HubApiModel {
-  async getStream (request: Request) {
-    console.log(request);
+  private _options: any;
 
-    const ret = new Readable({ read () {} });
-    ret.push({ foo: 'bar' });
-    ret.push(null);
-    return ret;
+  constructor (options: any) {
+    this._options = options;
+  }
+
+  getStream (request: Request) {
+    // TODO revisit how we get these search options
+    const searchOptions: ISearchApiOptions = request.app.locals.searchOptions;
+    const siteUrl = request.app.locals.hubSiteUrl || this._options.defaultSiteUrl;
+
+    const searchApiStream = new PagingStream({
+      firstPage: `${siteUrl}/api/v3/datasets?${getSearchQueryParams(searchOptions)}`,
+
+      loadPage: (url: string) => fetch(url).then(res => res.json()),
```

Comment:
question: can we use hub.js `searchContent` here vs having the v3 api url + bespoke query param construction in this project>

Reply:
Thanks for drawing my attention to this function. I'll need to evaluate what it currently supports and what we need for DCAT, but I'd like to reuse it in this context and throw away the bespoke stuff.

Reply:
fwiw - we have an issue extending `searchContent` so it can be passed either a site domain, or a catalog (if you've already got the site) and it will execute the search using that as the context... => https://devtopia.esri.com/dc/hub/issues/1475

I think that should streamline things for you, but I don't know all the details of what DCAT needs

Reply:
That would be perfect. Any idea on an ETA for that issue?

Reply:
Another thought is that I don't think the `searchContent` allows you to specify exactly which fields you want to have returned from the API. This wouldn't do anything for decreasing bandwidth in the Portal API context, but it's a pretty nice capability of the Hub API.

Reply:
Actually, it looks like it may already support this: https://esri.github.io/hub.js/api/search/IContentSearchOptions/#fields